rust/processor/src/worker.rs

```diff
@@ -262,95 +262,119 @@ impl Worker {
+                }
+                info!("[Parser] The stream is ended.");
+
+                // Don't try to reconnect if we've reached ending_version
+                if let Some(ending_version) = ending_version {
+                    if next_version_to_fetch > ending_version {
                         info!(
-                            processor_name = processor_name,
-                            start_version = start_version,
-                            end_version = end_version,
-                            channel_size = BUFFER_SIZE - tx.capacity(),
-                            channel_recv_latency_in_secs =
-                                last_insertion_time.elapsed().as_secs_f64(),
-                            "[Parser] Received chunk of transactions."
-                        );
-                        last_insertion_time = std::time::Instant::now();
-                    },
-                    Err(rpc_error) => {
-                        tracing::warn!(
                             processor_name = processor_name,
                             stream_address = indexer_grpc_data_service_address.clone(),
-                            error = ?rpc_error,
-                            "[Parser] Error receiving datastream response."
+                            ending_version = ending_version,
+                            "[Parser] Reached ending version.",
                         );
-                        if let Some(lrt) = last_reconnection_time {
-                            let elapsed = lrt.elapsed().as_secs();
-                            if elapsed < MIN_SEC_BETWEEN_GRPC_RECONNECTS {
-                                error!(
-                                    processor_name = processor_name,
-                                    stream_address = indexer_grpc_data_service_address.clone(),
-                                    seconds_since_last_retry = elapsed,
-                                    "[Parser] Recently reconnected. Will not retry.",
-                                );
-                                panic!("[Parser] Recently reconnected. Will not retry.")
-                            }
-                        }
-                        last_reconnection_time = Some(std::time::Instant::now());
-                        // Reconnecting
-                        tracing::warn!(
+                        break;
+                    }
+                }
+
+                // Handle reconnection logic
+                tracing::warn!(
+                    processor_name = processor_name,
+                    stream_address = indexer_grpc_data_service_address.clone(),
+                    "[Parser] Error receiving datastream response."
+                );
+                if let Some(lrt) = last_reconnection_time {
+                    let elapsed: u64 = lrt.elapsed().as_secs();
+                    if elapsed < MIN_SEC_BETWEEN_GRPC_RECONNECTS {
```

Comment:
is `MIN_SEC_BETWEEN_GRPC_RECONNECTS` defined in this file

Reply:
yep it's defined at 60 seconds at the top of this file